fsm/observer.go

```diff
@@ -46,9 +46,43 @@ func (c *CachedObserver) GetCachedNotifications() []Notification {
 	return c.cachedNotifications.Get()
 }
 
+// WaitForStateOption is an option for the WaitForState function.
+type WaitForStateOption struct {
+	initialWait time.Duration
+}
+
+// WithInitialWait sets the initial wait duration.
+func WithInitialWait(initialWait time.Duration) WaitForStateOption {
+	return WaitForStateOption{
+		initialWait: initialWait,
+	}
+}
+
 // WaitForState waits for the state machine to reach the given state.
+// If the optional initialWait parameter is set, the function will wait for
+// the given duration before checking the state. This is useful if the
+// function is called immediately after sending an event to the state machine
+// and the state machine needs some time to process the event.
 func (s *CachedObserver) WaitForState(ctx context.Context,
-	timeout time.Duration, state StateType) error {
+	timeout time.Duration, state StateType,
+	opts ...WaitForStateOption) error {
+
+	var initialWait time.Duration
+	if len(opts) > 0 {
+		for _, opt := range opts {
+			initialWait = opt.initialWait
+		}
+	}
+
+	// Wait for the initial wait duration if set.
+	if initialWait > 0 {
```

Comment:
Not sure this check is needed.

Reply:
Also nit on fix comment: https://github.com/lightninglabs/loop/pull/667/files#diff-28f5dffa528d8282177bcb103631cef19c6b2ae6d20abca5f559be88209469d6R129